presto-main/src/main/java/io/prestosql/operator/window/LagFunction.java

```diff
@@ -52,23 +59,15 @@ public void processRow(BlockBuilder output, int frameStart, int frameEnd, int cu
             long offset = (offsetChannel < 0) ? 1 : windowIndex.getLong(offsetChannel, currentPosition);
             checkCondition(offset >= 0, INVALID_FUNCTION_ARGUMENT, "Offset must be at least 0");
 
-            long valuePosition;
+            long valuePosition = currentPosition - offset;
 
-            if (ignoreNulls && (offset > 0)) {
-                long count = 0;
-                valuePosition = currentPosition - 1;
-                while (withinPartition(valuePosition, currentPosition)) {
-                    if (!windowIndex.isNull(valueChannel, toIntExact(valuePosition))) {
-                        count++;
-                        if (count == offset) {
-                            break;
-                        }
-                    }
-                    valuePosition--;
+            if (ignoreNulls && withinPartition(valuePosition, currentPosition) && offset > 0) {
+                if (windowIndex.isNull(valueChannel, toIntExact(valuePosition))) {
+                    valuePosition = lastNonNull;
+                }
+                else {
+                    lastNonNull = valuePosition;
```

Comment:
I think it isn't correct

for `lag(3)` if you check 3 positions abck and this happens to be a non-null value, you do not know whether it's 3-non-null values back. It may be the previous non null (`lag(1)`)

do we have some tests that would excercise this scenario?